web_src/js/features/codeeditor.ts

```diff
@@ -101,6 +114,25 @@ export async function createMonaco(textarea: HTMLTextAreaElement, filename: stri
       'focusBorder': '#0000', // prevent blue border
     },
   });
+}
+
+type CreateMonacoOpts = MonacoOpts & {language?: string};
+
+export async function createMonaco(textarea: HTMLTextAreaElement, filename: string, opts: CreateMonacoOpts) {
+  const monaco = await import(/* webpackChunkName: "monaco" */'monaco-editor');
+
+  initLanguages(monaco);
+  let {language, ...other} = opts;
+  if (!language) language = getLanguage(filename);
+
+  const container = document.createElement('div');
+  container.className = 'monaco-editor-container';
+  textarea.parentNode?.append(container);
```

Comment:
I guess we can throw if `parentNode` is absent. Throw can be achieved by either `!` or explicit throw.